source/extensions/filters/network/http_connection_manager/config.h

```diff
@@ -245,6 +246,7 @@ class HttpConnectionManagerConfig : Logger::Loggable<Logger::Id::config>,
   HttpConnectionManagerProto::ServerHeaderTransformation server_transformation_{
       HttpConnectionManagerProto::OVERWRITE};
   std::string server_name_;
+  absl::optional<std::string> scheme_;
```

Comment:
nit

```suggestion
  absl::optional<std::string> scheme_to_set_;
```

(Or `scheme_to_overwrite_` which might be more clear, up to you)